ops/internal/manage/collect.go

```diff
@@ -92,3 +93,79 @@ func CollectChainConfigs(p string) ([]DiskChainConfig, error) {
 
 	return out, nil
 }
+
+// FindChainConfig searches all superchains for given chain ID
+func FindChainConfig(wd string, chainId uint64) (*DiskChainConfig, config.Superchain, error) {
+	superchains, err := paths.Superchains(wd)
+	if err != nil {
+		return nil, "", fmt.Errorf("error getting superchains: %w", err)
+	}
+
+	for _, superchain := range superchains {
+		cfgs, err := CollectChainConfigs(paths.SuperchainDir(wd, superchain))
+		if err != nil {
+			return nil, "", fmt.Errorf("error collecting chain configs: %w", err)
+		}
+
+		for _, cfg := range cfgs {
+			if cfg.Config.ChainID == chainId {
+				return &cfg, superchain, nil
+			}
+		}
+	}
+
+	return nil, "", fmt.Errorf("chain with id %d not found", chainId)
+}
+
+type ChainConfigTuple struct {
+	Config     *DiskChainConfig
+	Superchain config.Superchain
+}
+
+// FindChainConfigs searches all superchains for the given chain IDs
+// - returns an error if any chain ID is not found
+func FindChainConfigs(wd string, chainIds []uint64) ([]ChainConfigTuple, error) {
```

Comment:
I'd rather modify the collect method to return the superchain data then have the caller do the filtering as opposed to passing in chain IDs to filter on here. We should only be walking the disk once.

Reply:
Good call. I removed `FindChainConfigs` in https://github.com/ethereum-optimism/superchain-registry/pull/951/commits/de46b1a8fb7c8e7d1693c08351753fba87b73b6b. The `[]DiskChainConfigs` returned by `CollectChainConfigs` now includes the `Superchain` for each config